src/entrance.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+
+import { PortalConsumer, PortalValue } from './context';
+
+interface EntrancePortalProps {
+  name: string;
+  children?: React.ReactNode;
+}
+
+interface EntranceProps extends EntrancePortalProps {
+  context: PortalValue;
+}
+
+class Entrance extends React.PureComponent<EntranceProps> {
```

Comment:
I don't feel strongly about this in the studio code base, but I'd rather see these as function components with hooks. The React team has stated that they expect the future of React to be hooks (see [this FAQ answer](https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both)). Makes it a little more friendly for open-sourcing, I think.

Should be able to do the work you're doing in the lifecycles here with `useEffect`

Something like:

```ts
React.useEffect(
  () => {
    props.context.setPortal(props.name, props.children);
    // Return an "unsubscriber" function
    return () => {
      props.context.setPortal(props.name, null);
    };
  },
  [props.name, props.children, props.context]
);
```

Reply:
Yeah I was thinking about using functional components and hooks but fell back to class components, will switch it now!